src/contexts/alert_handler.jsx

```diff
@@ -0,0 +1,38 @@
+import React, { createContext, useContext, useState } from "react";
+
+const AlertHandlerContext = createContext();
+
+export const AlertHandlerProvider = ({ children }) => {
+  const [alertState, setAlertState] = useState({
+    visible: false,
+    errorText: "",
+  });
+
+  const setAlert = (errorText) => {
+    setAlertState((prevAlertState) => ({
+      ...prevAlertState,
+      visible: true,
+      errorText: errorText,
+    }));
+  };
+
+  const closeAlert = () => {
+    const alertStateCopy = Object.assign({}, alertState);
+    alertStateCopy.visible = false;
+    setAlertState(alertStateCopy);
```

Comment:
```suggestion
    setAlertState({...alertState, visible: false});
```
[スプレッド構文](https://developer.mozilla.org/ja/docs/Web/JavaScript/Reference/Operators/Spread_syntax)を使うとこのようにスッキリとかけます。
Reactにおいては、変数やオブジェクトの値を書き換える（mutate）操作には注意が必要なので、visible二代入するよりも、visibleをfalseにした新たなオブジェクトを生成するほうが良い書き方と言えます。

参考：https://zenn.dev/canalun/articles/immutability_for_go_and_react